lint/yodaStyleExpr_checker.go

```diff
@@ -0,0 +1,55 @@
+package lint
+
+//! Detects Yoda style expressions that suggest to replace them.
+//
+// @Before:
+// if nil != ptr {}
+//
+// @After:
+// if ptr != nil {}
+
+import (
+	"go/ast"
+	"go/token"
+)
+
+func init() {
+	addChecker(&yodaStyleExprChecker{})
+}
+
+type yodaStyleExprChecker struct {
+	checkerBase
+}
+
+func (c *yodaStyleExprChecker) VisitLocalExpr(expr ast.Expr) {
+	binexpr, ok := expr.(*ast.BinaryExpr)
+	if !ok {
+		return
+	}
+	if binexpr.Op == token.EQL || binexpr.Op == token.NEQ {
+		if c.isConstExpr(binexpr.X) && c.isVar(binexpr.Y) {
```

Comment:
Why not `c.isConstExpr(binexpr.X) && !c.isConstExpr(binexpr.Y)`?
Would it break something?

Reply:
Or is it a precaution to avoid suggestions in a potentially long (multi-line) expressions?
Would make sense.. But would give less valid warnings as well.